### ISSUE TITLE
[Bug]: Sending money to MPESA transaction fails when using Equity bank

### ISSUE THREAD

**Issue by jnjerin** [UseHover/Stax#909]
### Description

Transaction fails when sending money to MPESA with Equity bank returning a FAIL status message stating no response from Equity bank.

### Steps to reproduce

1. Open Stax app.
2. On the home screen click on the “send money” icon.
3. On the “pay with” input field, select the Equity bank account.
4. On the “recipient network” input field, select MPESA.
5. On the “Contact phone number” input field, enter your phone number.
6. On the “Amount” input field, enter the amount you wish to send and click continue.
7. When prompted with the “Finish importing account” modal click connect now.
8. Select SIM to use and enter PIN.
9. Observe the transaction process.

### Expected result

The transaction completes successfully after prompting PIN, recipient receives money and returns a SUCCESS status message.

### Actual result

The transaction proceeds to complete, returns a FAIL status message and changes transaction type to 'check balance'

### Link to screenshots/screen recordings

https://drive.google.com/file/d/1tSZFmVsZ1Y3CMc7NiDzQBLi0dZagCFwV/view?usp=sharing

### Device name

Redmi Note 9S

### Android OS

10

### App version

1.18.10

### Config version

20230106114514

### Is your phone dual SIM or single SIM?

Dual SIM

### Which telco are you using for this test?

Safaricom

### Stax device ID or email address

11f9c800e06a3cea

**Comment by davkutalek** [comment]
@jnjerin we run a check balance on your bank first so that you can choose which account to send money from if you have more than one. We need to improve how it handles the failure case where we don't manage to check your balances, but I think if you restart your device the balance check will succeed. The reason you need to restart is likely due to the crash we were having in the previous update, this seems to trigger a bug in Android itself which is solved by the reboot.

**Comment by jnjerin** [comment]
@davkutalek you're right. This sort of worked. The transactions process successfully intermittently after restart. After some time it defaults back to this error then I have to restart again.
